### DISCUSSION TITLE
Branch tag history

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#3843] [Q&A]
Is there a way, either with `obslog` or (more hackily) `op log`, to see the history of a branch over time? I could probably reconstruct it by looking for `branch set` commands in `op log`, as well as `new` and `commit` (since I use the auto-branch-advancing feature), but it seems like this is something that `jj` should be able to determine more naturally.

**Comment by yuja** [comment]
I don't think there's a command to show ref history, but https://github.com/martinvonz/jj/pull/3617 is closer. If it's integrated with `jj op log`, maybe we can do something like `jj op log --revision branch_name --summary` to filter operations by `branch_name` change and show diff summary for each operation.
